run.go

```diff
@@ -332,178 +331,71 @@ func Log(c chan *SimulationStatus) DomainManipulator {
-		}
-	}
-	return result
-}
-
-// checkForDerivitives identifies the unique input variables that are required
-// to calculate the requested output variables.
-// Inputs:
-// (1) Map of requested output variable names to their corresponding expressions.
-// (2) Map of all function names to function definitions that are used in expressions.
-// Outputs:
-// (1) Map of output variable names to revised expressions where any user-defined
-// output variable showing up in a subsequent expression is replaced by its
-// corresponding user-defined expression.
-// (2) Slice of all unique input variables required to calculate the requested
-// output variables.
-func checkForDerivitives(m map[string]string, f map[string]govaluate.ExpressionFunction) (map[string]string, []string, error) {
-	getVariables := make([]string, 0, len(m))
-	for key, val := range m {
-		expression, err := govaluate.NewEvaluableExpressionWithFunctions(val, f)
-		if err != nil {
-			return nil, nil, fmt.Errorf("inmap OutputVariables: %v", err)
-		}
-		uniqueVars := removeDuplicates(expression.Vars())
-		getVariables = append(getVariables, uniqueVars...)
-		for _, uniqueVar := range uniqueVars {
-			if m[uniqueVar] != "" && m[uniqueVar] != uniqueVar {
-				m[key] = strings.Replace(m[key], uniqueVar, "("+m[uniqueVar]+")", -1)
-				return checkForDerivitives(m, f)
-			}
-		}
-	}
-	return m, removeDuplicates(getVariables), nil
-}
-
-// checkGetNames checks whether the unique input variables required to calculate
-// the user-requested output variables are available in the model.
-func (d *InMAP) checkGetNames(g ...string) error {
-	tempOutputNames, _, _ := d.OutputOptions()
-	outputNames := make(map[string]uint8)
-	for _, n := range tempOutputNames {
-		outputNames[n] = 0
-	}
-	for _, v := range g {
-		if _, ok := outputNames[v]; !ok {
-			return fmt.Errorf("inmap: unsupported variable name '%s'", v)
-		}
-	}
-	return nil
-}
-
-// checkOutputNames checks (1) if any output variable names exceed 10 characters
-// and (2) if any output variable names include characters that are unsupported
-// in shapefile field names.
-func (d *InMAP) checkOutputNames(o map[string]string) error {
-	for key := range o {
-		long := len(key) > 10
-		noCharError, err := regexp.MatchString("^[A-Za-z]\\w*$", key)
-		if err != nil {
-			panic(err)
-		}
-		if long && !noCharError {
-			return fmt.Errorf("inmap: output variable name '%s' exceeds 10 characters and includes unsupported character(s)", key)
-		} else if long {
-			return fmt.Errorf("inmap: output variable name '%s' exceeds 10 characters", key)
-		} else if !noCharError {
-			return fmt.Errorf("inmap: output variable name '%s' includes unsupported characters", key)
-		}
-	}
-	return nil
-}
-
 // Results returns the simulation results.
 // Output is in the form of map[variable][row]concentration.
-// If allLayers is true, the function returns data for all of the vertical
+// If AllLayers is true, the function returns data for all of the vertical
```

Comment:
We should move the documentation that is specific to Outputter to where the Outputter struct is defined.